src/audio/sound_buffer.rs

```diff
@@ -99,6 +100,11 @@ impl SoundBufferRef {
         unsafe { ffi::sfSoundBuffer_getSampleCount(self.raw()) as i64 }
     }
 
+    /// Get the samples stored in the buffer
+    pub fn get_samples(&self) -> &[i16] {
+        unsafe { slice::from_raw_parts(ffi::sfSoundBuffer_getSamples(self.raw()), self.sample_count() as usize) }
```

Comment:
This can potentially cause undefined behavior on platforms where `usize` is 32 bits and `sample_count()` returns a value greater than u32::MAX. Perhaps we should check for this condition.

Reply:
So if the `sample_count` is bigger than `u32::max_value` what should happen? should it return and `Option<&[i16]>`?

Reply:
Or should I just cap it to `u32::max_value()` length

Reply:
Since this is such an unlikely scenario, I think it's appropriate to simply panic here.

Reply:
If it was stable, we could use the [TryFrom](https://doc.rust-lang.org/nightly/std/convert/trait.TryFrom.html) API, which would be easier than manually implementing this.